### PR TITLE
executor: fix a bug for ADMIN SHOW DDL JOB QUERIES

### LINKED ISSUE #42298: ADMIN SHOW DDL JOB QUERIES return wrong results
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```
mysql> ADMIN SHOW DDL JOB QUERIES LIMIT 1;
+--------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| JOB_ID | QUERY                                                                                                                                                                                                                                                                                                                                                                |
+--------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 92     | ALTER TABLE `test`.`multi_indexes` ADD UNIQUE KEY `uniq_subaccount_transaction`(`f_subaccount_id`,`f_transaction_type`,`f_transaction_id`), ADD KEY `idx_user_id`(`f_user_id`), ADD KEY `idx_account_state_subaccount`(`f_account_id`,`f_state`,`f_subaccount_id`), ADD KEY `idx_state_precision`(`f_state`,`f_precision`), ADD KEY `idx_updated_at`(`f_updated_at`) |
+--------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)

mysql> ADMIN SHOW DDL JOB QUERIES LIMIT 50;
Empty set (0.01 sec)

```

### 2. What did you expect to see? (Required)

`ADMIN SHOW DDL JOB QUERIES LIMIT 50` should return at least one result.

### 3. What did you see instead (Required)
`ADMIN SHOW DDL JOB QUERIES LIMIT 50` return empty set.

The root cause is 
https://github.com/pingcap/tidb/blob/8aa4a899a3797cc52b4536cce4dead431028cef8/executor/executor.go#L802.

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
v6.6.0

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What problem does this PR solve?

Issue Number: close #42298

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix a bug for ADMIN SHOW DDL JOB QUERIES
```

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -6481,3 +6481,17 @@ from ssci right join csci on (ssci.customer_sk=csci.customer_sk
                                and ssci.item_sk = csci.item_sk)
 limit 100;`)
 }
+
+func TestIssue42298(t *testing.T) {
```
Comment (hawkingrei) [inline executor/executor_test.go:6485]: It should be put into ```executor/issuetest``` or append the existed test cases. ```executor``` has too many test cases.

**Review by lance6716 — APPROVED** [review]
should address hawkingrei's comment before merging

File: executor/executor_test.go (hunk 1 above)
Comment (CbcWestwolf) [reply executor/executor_test.go]: Done. PTAL

**Comment by lance6716** [comment]
ping @hawkingrei

**Review by gozssky — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
> ping @hawkingrei

LGTM

**Pull request merged**